src/Splitters/HtmlSplitter.php

```diff
@@ -23,29 +24,117 @@ class HtmlSplitter implements SplitterContract
      *
      * @var string[]
      */
-    protected $tags = [
+    protected $acceptedNodes = [
         'h1',
         'h2',
         'h3',
         'h4',
         'h5',
+        'h6',
         'p',
     ];
 
     /**
      * Creates a new instance of the class.
      *
-     * @param array $tags
+     * @param array $acceptedNodes
      *
      * @return void
      */
-    public function __construct(array $tags = null)
+    public function __construct(array $acceptedNodes = null)
     {
-        if ($tags !== null) {
-            $this->tags = $tags;
+        if ($acceptedNodes !== null) {
+            $this->acceptedNodes = $acceptedNodes;
         }
     }
 
+    /**
+     * Find it's value in $acceptedNodes.
+     *
+     * @param array $object
+     *
+     * @return int
+     */
+    public function findValue($object): int
+    {
+        return (int) array_search((key($object)), $this->acceptedNodes);
+    }
+
+    /**
+     * Add object to queue.
+     *
+     * @param array $object
+     * @param array $queue
+     *
+     * @return array
+     */
+    public function addObjectToQueue($object, $queue): array
+    {
+        if (count($queue) == 0) {
+            $queue[] = $object;
+
+            return $queue;
+        } else {
+            if ($this->findValue($object) > $this->findValue(end($queue))) {
+                $queue[] = $object;
+
+                return $queue;
+            } else {
+                array_pop($queue);
+
+                return $this->addObjectToQueue($object, $queue);
+            }
+        }
+    }
+
+    /**
+     * Importance formula.
+     * Give integer from tags ranking.
+     *
+     * @param \DOMElement $node
+     * @param array $queue
+     *
+     * @return int
+     */
+    public function importanceWeight($node, $queue): int
+    {
+        if ($node->nodeName == 'p') {
+            if (empty(end($queue))) {
+                return 0;
+            }
+
+            return (int) (count($this->acceptedNodes) - 1) + (int) (array_search(key(end($queue)), $this->acceptedNodes));
+        }
+
+        return (int) array_search($node->nodeName, $this->acceptedNodes);
+    }
+
+    /**
+     * Clean Records to have a correct format.
+     *
+     *
+     * @param array $records
+     *
+     * @return array
+     */
+    public function cleanRecords($records): array
+    {
+        $newRecords = [];
+        foreach ($records as $record) {
+            foreach ($record as $r) {
+                foreach ($r as $res => $values) {
+                    $newRecord[$res] = $values;
+                    if ($res == 'importance') {
```

Comment:
Missing const here for `importance`. Please review the PR for finding others cases here the usage of `const` is missing.